src/main/java/com/yahoo/memory/AllocateDirect.java

```diff
@@ -35,7 +35,9 @@ private AllocateDirect(final ResourceState state) {
   }
 
   static AllocateDirect allocate(final ResourceState state) {
-    state.putNativeBaseOffset(unsafe.allocateMemory(state.getCapacity()));
+    final long capacity = state.getCapacity();
+    state.putNativeBaseOffset(unsafe.allocateMemory(capacity));
+    NioBits.reserveMemory(capacity);
```

Comment:
Shouldn't reserveMemory go before unsafe.allocateMemory(), to give a chance to System.gc()?

Reply:
Done.